File: backend/central_learning/src/main/java/aicon/lifehack/central_learning/model/Course.java
package aicon.lifehack.central_learning.model;


import java.util.Date;

import lombok.Data;

@Data
public class Course {
    private String course_id;
    // This is our "foreign key" relationship
    private String topic_id;
    private String title;
    private String description;
    private String author;
    private Date created_at;
    private int like_count = 0;
    // You could add more fields here later, like 'difficulty', 'imageUrl', etc.
}

File: frontend/models/Course.ts
import { Lesson } from "./Lesson";

export interface Course {
  course_id: string;
  topic_id: string;
  title: string;
  description: string;
  author: string;
  date: string;
  like_count?: number; // Default to 0 when used
  lessons?: Lesson[]; // Array of lessons
}
